runtime/src/test/java/com/vaadin/quarkus/QuarkusInstantiatorTest.java

```diff
@@ -0,0 +1,263 @@
+import org.mockito.Mockito;
+
+@QuarkusTest
+public class QuarkusInstantiatorTest {
+
+    @Singleton
+    public static class SomeCdiBean {
+    }
+
+    @ApplicationScoped
+    public static class ScopedCdiBean {
+    }
+
+    @Interceptor
+    @MyBinding
+    public static class MyInterceptor {
+
+        @AroundInvoke
+        public Object intercept(InvocationContext ctx) throws Exception {
+            return ctx.proceed();
+        }
+    }
+
+    @InterceptorBinding
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target({ ElementType.TYPE, ElementType.METHOD })
+    public @interface MyBinding {
+
+    }
+
+    @Singleton
+    public static class InterceptedCdiBean {
+
+        @MyBinding
+        public void exec() {
+
+        }
+    }
+
+    public static class ParentBean {
+
+        @Inject
+        private BeanManager bm;
+
+        public BeanManager getBm() {
+            return bm;
+        }
+
+    }
+
+    @Singleton
+    @Tag("div")
+    public static class SingletonComponent extends Component {
+        @Inject
+        BeanManager beanManager;
+    }
+
+    @Vetoed
+    @Singleton
+    public static class NotACdiBean extends ParentBean {
+    }
+
+    public static class Ambiguous extends ParentBean {
+    }
+
+    @VaadinServiceEnabled
+    @Singleton
+    public static class I18NTestProvider implements I18NProvider {
+
+        @Override
+        public List<Locale> getProvidedLocales() {
+            return null;
+        }
+
+        @Override
+        public String getTranslation(String key, Locale locale,
+                Object... params) {
+            return null;
+        }
+
+    }
+
+    @Singleton
+    public static class ServiceInitObserver {
+
+        ServiceInitEvent event;
+
+        public void serviceInit(@Observes ServiceInitEvent event) {
+            this.event = event;
+        }
+
+        public ServiceInitEvent getEvent() {
+            return event;
+        }
+
+    }
+
+    @Inject
+    private BeanManager beanManager;
+
+    @Inject
+    @VaadinServiceEnabled
+    private QuarkusInstantiatorFactory instantiatorFactory;
+
+    private Instantiator instantiator;
+
+    @Inject
+    private SomeCdiBean singleton;
+
+    @Inject
+    private ServiceInitObserver serviceInitObserver;
+
+    private ServiceUnderTestContext serviceUnderTestContext;
+
+    @BeforeEach
+    public void setUp() {
+        serviceUnderTestContext = new ServiceUnderTestContext(beanManager);
+        serviceUnderTestContext.activate();
+        instantiator = instantiatorFactory
+                .createInstantitor(VaadinService.getCurrent());
+    }
+
+    @AfterEach
+    public void tearDown() {
+        serviceUnderTestContext.tearDownAll();
+    }
+
+    @Test
+    public void getI18NProvider_beanEnabled_instanceReturned() {
+        I18NProvider i18NProvider = instantiator.getI18NProvider();
+        Assertions.assertNotNull(i18NProvider);
+        Assertions.assertTrue((i18NProvider instanceof I18NTestProvider));
```

Comment:
```suggestion
        Assertions.assertTrue(i18NProvider instanceof I18NTestProvider);
```
nit: some extra parentheses